test/controllers/api/v2/content_views_controller_test.rb

```diff
@@ -170,6 +171,23 @@ def test_update_components
       assert_template 'katello/api/v2/common/update'
     end
 
+    def test_duplicate_component_error_message
+      view = katello_content_views(:library_view)
+      view_version = create(:katello_content_view_version, :content_view => view)
+
+      composite = katello_content_views(:composite_view)
+      ContentViewComponent.create!(:composite_content_view => composite,
+                                   :content_view_version => view_version, :latest => false)
+
+      view_version2 = create(:katello_content_view_version, :content_view => view)
+
+      put :update, params: { id: composite.id, content_view: { component_ids: [view_version.id, view_version2.id] } }
+      display_message = JSON.parse(response.body)['displayMessage']
+      test_message = "Validation failed: Content view components Another component already includes content view with ID " + view_version.content_view_id.to_s + "."
```

Comment:
You can use string interpolation (http://ruby-for-beginners.rubymonstas.org/bonus/string_interpolation.html) to make this cleaner:

```
expected = "Validation failed: Content view components Another component already includes content view with ID #{view_version.content_view_id}."

```